### PR TITLE
Add other init systems support

### LINKED ISSUE #31: Add other init systems support
Init systems we want to support:
- OpenRC
- Runit

### PR DESCRIPTION
**Pull request by cooligus**
Closes #31

### CODE REVIEW

File: src/api/daemon.go [hunk 1]
```diff
@@ -33,9 +34,78 @@ func getSystemdDaemon(daemonName string) (Daemon, error) {
 		Name:      daemonName,
 		IsActive:  isActive,
 		IsEnabled: isEnabled,
+		InitLevel: "", // TODO
 	}, nil
 }
 
+func getOpenRCDaemon(ctx context.Context, daemonName string) (Daemon, error) {
+	cmd := exec.CommandContext(ctx, "rc-service", daemonName, "status")
```
Comment (BaderBC) [inline src/api/daemon.go:42]: Do you remember how sql injection works?
For me this code looks like great way to make "command injection"

File: src/api/daemon.go [hunk 2]
```diff
@@ -33,9 +34,78 @@ func getSystemdDaemon(daemonName string) (Daemon, error) {
...
+		return Daemon{}, err
+	}
+
+	switch initSystem {
+	case SYSTEMD:
+		return getSystemdDaemon(ctx, daemonName)
+	case OPENRC:
+		return getOpenRCDaemon(ctx, daemonName)
+	case RUNIT:
+		return Daemon{}, err
```
Comment (BaderBC) [inline src/api/daemon.go:107]: 1.  `err` could be potentially `nil`
2. Error should include message that RUNIT is unimplemented

File: src/api/daemon.go [hunk 3]
```diff
@@ -33,9 +34,78 @@ func getSystemdDaemon(daemonName string) (Daemon, error) {
...
+
+	daemon.Name = daemonName
+
+	return daemon, nil
+}
+
 func GetDaemon(daemonName string) (Daemon, error) {
-	return getSystemdDaemon(daemonName)
+	initSystem, err := GetInitSystem()
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
```
Comment (BaderBC) [inline src/api/daemon.go:94]: What happens in case of time out?

File: src/api/daemon.go (hunk 1 above)
Comment (cooligus) [reply src/api/daemon.go:42]: How would you implement it then?

File: src/api/daemon.go (hunk 3 above)
Comment (cooligus) [reply src/api/daemon.go:94]: Great question: I don't know

File: src/api/daemon.go (hunk 2 above)
Comment (cooligus) [reply src/api/daemon.go:107]: Runit is going to be implemented in this PR so let me resolve this comment

File: src/api/daemon.go (hunk 1 above)
Comment (BaderBC) [reply src/api/daemon.go:42]: I would check if the name contains any invalid name characters like e.g.
whitespace characters, `;`, etc. (I'm not sure but I guess `;` is an illegal daemon name character)

File: src/api/daemon.go (hunk 3 above)
Comment (cooligus) [reply src/api/daemon.go:249]: It's handled now

**Review by sewe2000 — APPROVED** [review]

**Review by BaderBC — CHANGES_REQUESTED** [review]

**Review by BaderBC — APPROVED** [review]
LGTM!

**Pull request merged**